File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string chars;

        for (auto i: s) {
            if (isalnum(i)) {
                chars += tolower(i);
            }
        }

        int left = 0, right = chars.size() - 1;
        while (left <= right) {
            if (chars[left] != chars[right]) {
                return false;
            }
            left++;
            right--;
        }

        return true;
    }
};

File: LeetCode/Easy/0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        start = 0
        end = len(s) - 1

        while start < end:
            left_ch = s[start]
            right_ch = s[end]

            if s[start].isalnum():
                left_ch = left_ch.lower()
            else:
                start += 1
                continue

            if s[end].isalnum():
                right_ch = right_ch.lower()
            else:
                end -= 1
                continue

            if left_ch == right_ch:
                start += 1
                end -= 1
            else:
                return False

        return True